deepfence_worker/tasks/scans/delete_cloud_accounts.go

```diff
@@ -0,0 +1,283 @@
+	if err := deleteScans(ctx, accountID); err != nil {
+		log.Error().Err(err).Msgf("failed to delete scans for account %s", accountID)
+		return err
+	}
+	return deleteCloudResourceAndNode(ctx, accountID)
+
+}
+
+func deleteScans(ctx context.Context, accountID string) error {
+
+	ctx, span := telemetry.NewSpan(ctx, "scans", "delete-scans")
+	defer span.End()
+
+	log := log.WithCtx(ctx)
+
+	// delete Cloud/ComplianceScan's related to the account first
+	nodeIDs := []model.NodeIdentifier{{NodeID: accountID, NodeType: "cloud_account"}}
+	filters := reporters.FieldsFilters{}
+	window := model.FetchWindow{Offset: 0, Size: 10000000}
+
+	scans, err := reportersScan.GetScansList(ctx, utils.NEO4JCloudComplianceScan, nodeIDs, filters, window)
+	if err != nil {
+		log.Error().Err(err).Msgf("failed to list scans for cloud node %s", accountID)
+	}
+
+	defer log.Info().Msgf("deleted %d scans for account %s", len(scans.ScansInfo), accountID)
+
+	for _, s := range scans.ScansInfo {
+		err := reportersScan.DeleteScan(ctx, utils.NEO4JCloudComplianceScan, s.ScanID, []string{})
+		if err != nil {
+			log.Error().Err(err).Msgf("failed to delete scan id %s", s.ScanID)
+		}
+	}
+
+	return nil
+}
+
+func deleteCloudResourceAndNode(ctx context.Context, accountID string) error {
+
+	ctx, span := telemetry.NewSpan(ctx, "scans", "delete-cloud-resources-and-node")
+	defer span.End()
+
+	log := log.WithCtx(ctx)
+
+	defer log.Info().Msgf("deleted cloud node and resources for account %s", accountID)
+
+	driver, err := directory.Neo4jClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	session := driver.NewSession(ctx, neo4j.SessionConfig{AccessMode: neo4j.AccessModeWrite})
+	defer session.Close(ctx)
+
+	tx, err := session.BeginTransaction(ctx, neo4j.WithTxTimeout(30*time.Second))
+	if err != nil {
+		return err
+	}
+	defer tx.Close(ctx)
+
+	// delete cloud node and resources
+	deleteQuery := `
+	MATCH (n:CloudNode{node_id: $node_id})-[:OWNS]->(r:CloudResource)
+	DETACH DELETE n,r
+	`
+
+	if _, err := tx.Run(ctx, deleteQuery, map[string]any{"node_id": accountID}); err != nil {
+		log.Error().Err(err).Msgf("failed to delete cloud node and resources for account %s", accountID)
+		return err
+	}
+
+	// required in case where link cloud resources task has not yet run
+	deleteQuery1 := `MATCH (n:CloudNode{node_id: $node_id})	DETACH DELETE n`
+	deleteQuery2 := `MATCH (r:CloudResource{account_id: $node_id}) DETACH DELETE r`
+
+	if _, err := tx.Run(ctx, deleteQuery1, map[string]any{"node_id": accountID}); err != nil {
+		log.Error().Err(err).Msgf("failed to delete cloud node account %s", accountID)
+		return err
+	}
+
+	if _, err := tx.Run(ctx, deleteQuery2, map[string]any{"node_id": accountID}); err != nil {
+		log.Error().Err(err).Msgf("failed to delete cloud resources for account %s", accountID)
+		return err
+	}
+
+	// delete hosts discovered from cloud
+	deleteHostsQuery := `
+	MATCH (n:Node{account_id: $node_id})
+	WHERE n.agent_running=false
+	DETACH DELETE n
+	`
```

Comment:
I think we should not delete the `Node`  directly and leave this job to the clean up task, there are couple of edge cases to be handled
Let's mark it as not running via:
```
MATCH (n:Node{account_id: $node_id})
SET n.agent_running=false
```
Instead